diff --git a/drivers/hid/wacom_sys.c b/drivers/hid/wacom_sys.c
index e06af5b..1ad489a 100644
--- a/drivers/hid/wacom_sys.c
+++ b/drivers/hid/wacom_sys.c
@@ -328,7 +328,7 @@ static int wacom_set_device_mode(struct hid_device *hdev, int report_id,
 	unsigned char *rep_data;
 	int error = -ENOMEM, limit = 0;
 
-	rep_data = kzalloc(length, GFP_KERNEL);
+	rep_data = kzalloc(length, GFP_NOIO);
 	if (!rep_data)
 		return error;
 
